src/main/java/roomescape/apply/reservationwaiting/application/WaitingPositionCalculator.java

```diff
@@ -0,0 +1,28 @@
+package roomescape.apply.reservationwaiting.application;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.apply.reservationwaiting.domain.ReservationWaiting;
+import roomescape.apply.reservationwaiting.domain.ReservationWaitingRepository;
+
+@Service
+@Transactional(readOnly = true)
+public class WaitingPositionCalculator {
+
+    private final ReservationWaitingRepository reservationWaitingRepository;
+
+    public WaitingPositionCalculator(ReservationWaitingRepository reservationWaitingRepository) {
+        this.reservationWaitingRepository = reservationWaitingRepository;
+    }
+
+    public long calculateByReservationWaiting(ReservationWaiting waiting) {
+        long waitingPosition = reservationWaitingRepository.countByThemeIdAndDateAndTimeIdAndWaitingTimeLessThan(
+                waiting.getTheme().getId(),
+                waiting.getReservationDate(),
+                waiting.getTime().getId(),
+                waiting.getWaitingTime()
+        );
+        return waitingPosition + 1;
```

Comment:
크게 중요한 부분은 아니지만 메서드 이름으로 코드의 의도를 조금 더 명확히 드러낼 수 도 있습니다 :) 
```suggestion
        return updateWaitingPositionStartFromOne(waitingPosition);
```

Reply:
이 부분도 생각해보니 어플리케이션에서는 0부터 시작하지만 사용자에게 보여주기 위한 값 변경이니 PresetationLayer에 두는 것이 어떨까요??